onnxruntime/core/optimizer/qdq_transformer/qdq_actions.cc

```diff
@@ -0,0 +1,220 @@
+  NTO::NodeLocation q{NTO::NodeType::kOutput, 0};
+
+  std::vector<NodeAndMoveInfo> moves{
+      MoveAndAppend(q, ArgType::kInput, 1, ArgType::kInput),   // append scale (input 1) from q
+      MoveAndAppend(q, ArgType::kInput, 2, ArgType ::kInput),  // append zp (input 2) from q
+      MoveAll(variadic_dq, ArgType::kInput),                   // append all inputs from all dq nodes
+      MoveAll(q, ArgType::kOutput)};                           // and use the outputs from q
+
+  return moves;
+}
+
+// moves for replacing a node with a Conv node with DQ inputs with the qlinear version
+std::vector<NodeAndMoveInfo> ConvMoves() {
+  NTO::NodeLocation dq_x{NTO::NodeType::kInput, 0};
+  NTO::NodeLocation dq_w{NTO::NodeType::kInput, 1};
+  NTO::NodeLocation dq_bias{NTO::NodeType::kInput, 2};
+  NTO::NodeLocation q{NTO::NodeType::kOutput, 0};
+
+  std::vector<NodeAndMoveInfo> moves{
+      MoveAll(dq_x, ArgType::kInput),                                     // append all inputs from x
+      MoveAll(dq_w, ArgType::kInput),                                     // append all inputs from w
+      MoveAndAppend(q, ArgType::kInput, 1, ArgType::kInput),              // append scale (input 1) from q
+      MoveAndAppend(q, ArgType::kInput, 2, ArgType ::kInput),             // append zp (input 2) from q
+      MoveAndAppend(dq_bias, ArgType::kInput, 0, ArgType::kInput, true),  // (optional) append bias
+      MoveAll(q, ArgType::kOutput)};                                      // and use the outputs from q
+
+  return moves;
+}
+
+QDQReplaceWithNew MatMulIntToFloatReplacer() {
+  NTO::NodeLocation dq1{NTO::NodeType::kInput, 0};
+  NTO::NodeLocation dq2{NTO::NodeType::kInput, 1};
+  NTO::NodeLocation target{NTO::NodeType::kTarget, 0};
+
+  std::vector<NodeAndMoveInfo> moves{
+      MoveAndAppend(dq1, ArgType::kInput, 0, ArgType::kInput),
+      MoveAndAppend(dq2, ArgType::kInput, 0, ArgType::kInput),
+      MoveAndAppend(dq1, ArgType::kInput, 1, ArgType::kInput),
+      MoveAndAppend(dq2, ArgType::kInput, 1, ArgType::kInput),
+      MoveAndAppend(dq1, ArgType::kInput, 2, ArgType::kInput),
+      MoveAndAppend(dq2, ArgType::kInput, 2, ArgType::kInput),
+      MoveAll(target, ArgType::kOutput)};
+
+  return QDQReplaceWithNew(kMSDomain, std::move(moves), "MatMulIntegerToFloat");
+}
+
+ReplaceWithQLinear MatMulQLinearReplacer() {
+  NTO::NodeLocation dq1{NTO::NodeType::kInput, 0};
+  NTO::NodeLocation dq2{NTO::NodeType::kInput, 1};
+  NTO::NodeLocation q{NTO::NodeType::kOutput, 0};
+
+  std::vector<NodeAndMoveInfo> moves{
+      MoveAll(dq1, ArgType::kInput),                           // append all inputs from dq to new node
+      MoveAll(dq2, ArgType::kInput),                           // append all inputs from dq to new node
+      MoveAndAppend(q, ArgType::kInput, 1, ArgType::kInput),   // append scale (input 1) from q
+      MoveAndAppend(q, ArgType::kInput, 2, ArgType ::kInput),  // append zp (input 2) from q
+      MoveAll(q, ArgType::kOutput)};
```

Comment:
is this BinaryMoves()?

Reply:
Nice catch! Yes it is. Can just use BinaryReplaceWithQLinear